core/tests/test_utils.py

```diff
@@ -824,9 +822,14 @@ def put_json(self, url, payload, csrf_token=None, expected_status_int=200):
         self.assertEqual(json_response.status_int, expected_status_int)
         return self._parse_json_response(json_response, expect_errors)
 
-    def get_csrf_token_from_response(self, response):
-        """Retrieve the CSRF token from a GET response."""
-        return re.search(CSRF_REGEX, response.body).group(1)
+    def get_csrf_token(self):
```

Comment:
This is security-related, so I think we should make sure to generate this the usual way (i.e. from the controllers). Can you redo this method so that it obtains the CSRF token through an AJAX request the way a normal page would (rather than "reaching in" to internal functions in the backend)?

Reply:
@seanlip, for some reason, doing it this way causes the tests to error.

Reply:
That could indicate a deeper underlying problem. Can you figure out why the tests are error-ing?

Reply:
Done